dftio/io/abacus/abacus_parser.py

```
@@ -134,9 +187,10 @@ def get_blocks(self, idx, hamiltonian=True, overlap=False, density_matrix=False)
         hamiltonian_dict, overlap_dict, density_matrix_dict = None, None, None
         sys = self.raw_sys[idx]
         nsites = sys.data["atom_types"].shape[0]
-        if os.path.exists(os.path.join(self.raw_datas[idx], "OUT.ABACUS", "hscsr.tgz")):
-            os.system(f"tar -xzf {os.path.join(self.raw_datas[idx], 'OUT.ABACUS', 'hscsr.tgz')} -C {os.path.join(self.raw_datas[idx])}")
-        with open(os.path.join(self.raw_datas[idx], "OUT.ABACUS", logfile), 'r') as f:
+        output_dir = self._get_output_dir(idx)
+        if os.path.exists(os.path.join(output_dir, "hscsr.tgz")):
+            os.system(f"tar -xzf {os.path.join(output_dir, 'hscsr.tgz')} -C {os.path.join(self.raw_datas[idx])}")
```

Ruff (0.14.5)

Starting a process with a shell, possible injection detected

(S605)